horust/src/horust/formats/service.rs

```diff
@@ -624,6 +629,52 @@ impl From<TerminationSignal> for Signal {
     }
 }
 
+#[derive(Serialize, Clone, Deserialize, Debug, Eq, PartialEq)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct Resource {
+    #[serde(default)]
+    /// The percentage of CPU quota that the process can use, negative value means no limit
+    pub(crate) cpu_percent: i64,
+    #[serde(default, skip_serializing, deserialize_with = "str_to_bytes")]
+    /// The maximum amount of memory that the process can use
+    pub(crate) memory: u64,
+}
+
+impl Default for Resource {
+    fn default() -> Self {
+        Resource {
+            cpu_percent: -1,
+            memory: 1024 * 1024 * 1024 * 1024, // set to an extreme large value of 1 TiB
```

Comment:
I think it would be cleaner to use Options for the fields, and we can dynamically add them to the constructor if they are defined, what do you think?

Reply:
Fixed.